### PR TITLE
FI-401 add revinclude provenance tests

### PR DESCRIPTION
**Pull request by czh-orz**
This adds tests for revInclude provenance searches. To test, run US Core 3.0.1 against https://api-v8-r4.hspconsortium.org/czhour4/data and select "Adams, Daniel X". The allergy intolerance sequence should have a provenance associated with it. The revinclude test should pass for AllergyIntolerance, but not for any other other sequences.

Alternatively, use the Community tests and the open endpoint https://api-v8-r4.hspconsortium.org/czhour4/open with patient-id `SMART-1288992`

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-401
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [ ] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [x] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: generator/uscore/uscore_generator.rb [hunk 1]
```diff
@@ -131,6 +132,29 @@ def create_authorization_test(sequence)
         sequence[:tests] << authorization_test
       end
 
+      def create_revinclude_test(sequence)
+        revinclude_test = {
+          tests_that: "A Server SHALL be capable of supporting the following _revincludes: #{sequence[:revincludes].join(',')}",
```
Comment (arscan) [inline generator/uscore/uscore_generator.rb]: This is a different naming convention than we've been using so far.  The name doesn't have the "SHALL" requirement -- that information should be in the description of the test that you get when you click 'more' in the UI.

File: generator/uscore/uscore_generator.rb [hunk 2]
```diff
@@ -131,6 +132,29 @@ def create_authorization_test(sequence)
...
+          index: sequence[:tests].length + 1,
+          link: 'https://www.hl7.org/fhir/search.html#revinclude'
+        }
+        first_search = find_first_search(sequence)
+        search_params = first_search.nil? ? "\nsearch_params = {}" : get_search_params(first_search[:names], sequence)
+        revinclude_test[:test_code] = search_params
+        sequence[:revincludes].each do |revinclude|
+          resource_type = revinclude.split(':').first.downcase
```
[the remark this answers is not in the document]
Comment (czh-orz) [reply generator/uscore/uscore_generator.rb]: Ended up just using resource_type.capitalize for the duplicated case so I could keep things a little simpler.

File: generator/uscore/uscore_generator.rb [hunk 3]
```diff
@@ -131,6 +132,29 @@ def create_authorization_test(sequence)
...
+          revinclude_test[:test_code] += %(
+                search_params['_revinclude'] = '#{revinclude}'
+                reply = get_resource_by_params(versioned_resource_class('#{sequence[:resource]}'), search_params)
+                assert_response_ok(reply)
+                assert_bundle_response(reply)
+                #{resource_type}_results = reply&.resource&.entry&.map(&:resource)&.select { |resource| resource.resourceType == '#{revinclude.split(':').first}' }
+                assert #{resource_type}_results.any?, 'No #{resource_type} resources were returned from this search'
```
[the remark this answers is not in the document]
Comment (czh-orz) [reply generator/uscore/uscore_generator.rb]: I changed select? to any? as suggested in the other PR which also resolves this issue.

File: generator/uscore/uscore_generator.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jammjammjamm) [reply generator/uscore/uscore_generator.rb]: Will that still be ok for resources with internal capitalization?

[the remark this answers is not in the document]
Comment (czh-orz) [reply generator/uscore/uscore_generator.rb]: No, didn't think about that. I changed it back to use the original value.

File: generator/uscore/uscore_generator.rb (hunk 1 above)
Comment (czh-orz) [reply generator/uscore/uscore_generator.rb]: Updated the description to be more like the other test descriptions.

**Comment by czh-orz** [comment]
I updated the description to not include the SHALL wording

**Pull request merged**